### PR TITLE
enabled CANCELACION state + minor test refactor

### PR DESCRIPTION
**Pull request by felipaoo**
- added `CANCELACION` state which is the state for transactions like "TRASPASO CANCELADO". This state marks the transaction as `failed`
- fixture `default_outcome_transaction` refactor

### CODE REVIEW

**Comment by matin** [comment]
@felipaoo  you should add comments and info in he README on when each state occurs

**Comment by matin** [comment]
@felipaoo puedes marcar como borrador hasta que está listo para revisar?

**Comment by felipaoo** [comment]
It is not ready because there is a problem with a package (flask-mongoengine) that needs `wheel` package to be installed at installation time, when doing `make install`. I don't know why it is failing now and not before. I was reading it could be solved by including `pyproject.toml` file to specify what tools needs to exist before build but it doesn't work. @matin have you ever use this file?

**Comment by matin** [comment]
@paolodamico any recommendations?

**Comment by paolodamico** [comment]
I noticed a couple of things, given that the project runs fine locally, and 2 out of the 3 tasks succeeded,
- The project should run on Python 3.7 according to the Makefile but Github actions runs on 3.8.
- The [actions config file](https://github.com/cuenca-mx/speid/blob/master/.github/workflows/test.yml) specifies the Python version as a string (L13) for the `lint` task (the failing one) and as a number for the others. I'm not sure of the implications of this, but I did notice some other differences in the task execution,
    - The `test` and `coverage` tasks run the `pip install wheel` command first before installing the dependencies on `requirements.txt` and `requirements-dev.txt`.
    - For some reason, the `actions/setup-python@v1` on `lint` explicitly sets a x64 architecture on the logs, which is not explicit on the other tasks.

**Recommendations to try:**
- Update [actions config file](https://github.com/cuenca-mx/speid/blob/master/.github/workflows/test.yml) so that all three tasks have the same parameters.
- Update the Makefile's `install-dev` command to run `pip install wheel` first.
- **[Bonus]** - Why not consider using something like tox so the project can run on multiple Python versions and increase support for third-parties using this library?

**Comment by felipaoo** [comment]
following @paolodamico's  recommendations: I will move `pip install wheel` inside `make install-test`. My first try was to run it directly on`test.yml`. 
there is a issue reporting this: https://github.com/MongoEngine/flask-mongoengine/issues/402

**Comment by paolodamico** [comment]
Glad it's working now!

**Comment by felipaoo** [comment]
Added `Transaction.detalle` because of this spec. It could help us to know the reason why a transfer failed.

image

File: tests/views/test_general.py [hunk 1]
```diff
@@ -18,110 +17,106 @@ def test_health_check(client):
...
-    client, default_outcome_transaction, mock_callback_queue
-):
-    trx = Transaction(**default_outcome_transaction)
-    trx.stp_id = DEFAULT_ORDEN_ID
-    trx.save()
-    id_trx = trx.id
 
-    data = dict(id=DEFAULT_ORDEN_ID, Estado='LIQUIDACION', Detalle="0")
+@pytest.mark.usefixtures('mock_callback_queue')
+def test_order_event_duplicated(client, outcome_transaction):
```
Comment (ricardo8990) [inline tests/views/test_general.py:97]: Creo que si vas a usar `pytest.mark.usefixtures` deberías usarlo para todos los fixtures, como `outcome_transaction`. 
```suggestion
@pytest.mark.usefixtures('mock_callback_queue', 'outcome_transaction')
def test_order_event_duplicated(client):
```

File: speid/views.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def health_check():
 def create_orden_events():
     try:
         transaction = Transaction.objects.get(stp_id=request.json['id'])
-
+        transaction.detalle = str(request.json['Detalle'])
```
Comment (ricardo8990) [inline speid/views.py:26]: Todos los callbacks vienen siempre con este campo? O solo cuándo son traspasos cancelados? De otra forma aquí podría dar un `KeyError` y no abonar las transferencias

File: tests/views/test_general.py (hunk 1 above)
Comment (felipaoo) [reply tests/views/test_general.py:97]: no podría usarlo como sugieres porque `outcome_transaction ` si se utiliza en el test mientras que `mock_callback_queue ` no

**Review by ricardo8990 — APPROVED** [review]

**Pull request merged**